### ISSUE TITLE
UI: clean cherrypick feedback modal presents as error

### ISSUE THREAD

**Issue by jrfnl** [gitextensions/gitextensions#9319]
## Current behaviour

When cherrypicking changes from another commit into the current branch, the UI always shows an error message, even when the cherrypicking was successful and applied cleanly.

## Expected behaviour

No error message when the patch applied cleanly.

## Steps to reproduce

* Have a branch with some changes to File A.
* Cherrypick the changes from File B (which hasn't been touched in the current branch and for which the changes would apply cleanly) from another branch.
* See the error display. 

## Screenshots

![image](https://user-images.githubusercontent.com/663378/124160681-cf251780-da9c-11eb-99b2-2c28813f8418.png)

## Did this work in previous version of GitExtensions

Yes, IIRC this did not happen prior to 3.5.0.

## Environment

- Git Extensions 3.5.0.11998
- Build 747591af7e0fa8bbacd241994db245b9e9c14266 (Dirty)
- Git 2.32.0.windows.1
- Microsoft Windows NT 10.0.19043.0
- .NET Framework 4.8.4360.0
- DPI 144dpi (150% scaling)

**Comment by jrfnl** [comment]
Note: this also happens when cherrypicking select lines from a commit using "Stage select lines".

IIRC there was a change in 3.5.0 where the menu options in the right click menu when lines in a file are highlighted in a commit in the diff view were renamed. Might this issue have something to do with those changes ?

**Comment by gerhardol** [comment]
Note: If this is Cherry-pick for the complete file in the diff file, this is not changed.
Similar for linepatching in the fileviewer (cherry-pick in 3.4, Stage in 3.5).
If there is a message from the Git command, it is printed.
Maybe this should be suppressed, #9056 changes the handling here.

Otherwise the handling for line patches has been revised some. It is possible to stage/unstage lines also in the Diff tab, comparing a commit to worktree artificial commit normally failed before.
(The handling in FormCommit was aligned with the other handling too.)

**Comment by jrfnl** [comment]
Well, what changed in my experience, is that previously on success, it would silently stage the changes, while now - even on success - it shows an **error** modal. Which is confusing at best.

**Comment by gerhardol** [comment]
GE is not changed. Try the portable for 3.4. Maybe a Git change. 

You may rephrase this to a feature request instead

**Comment by gerhardol** [comment]
This should be changed after #9056

**Comment by jrfnl** [comment]
@gerhardol Nice! looking forward to having a play with it.

**Comment by gerhardol** [comment]
@jrfnl Please review the PR
